### DISCUSSION TITLE
Is `of` really needed?

### DISCUSSION THREAD

**Discussion by senhalil** [yoav-lavi/melody#23] [Ideas]
First of all, amazing idea! Always wanted to write regexp like a normal language, thanks for this awesome project.

It looks like nobody asked so I figure I might as well be that person. I feel like `2 <matches>` is more expressive than `2 of match`. (Not sure what would entail and/or if it would be possible but here are my 2 cents.

**Comment by yoav-lavi** [comment]
Thank you for the kind words!

Regarding `of`, waiting to see what other people think, but I personally see it as a sort of link between the number and what it's operating on.

Syntactically it makes more sense to me to have `2 * 5` or `5.times { ... }` rather than `2(5)` or `5 { ... }`. 

Will consider though!

**Reply by senhalil** [reply]
Thanks for the answer. I do understand your point of view. 

I guess, I treated the syntax as ordinary English (which might not be expressive enough to handle all intricacies of regexp after all :thinking:) and the class code example (i.e., `// classname 1xx`) read more nicely in my head: 
```sh
some of <word>;
<space>;
"1";
2 of <digit>;
```
```sh
some <words>;
<space>;
"1";
2 <digits>;
```
Cheers!

**Reply by yoav-lavi** [reply]
I see, although that would require a plural form for each type of character (and that may not work for literals, e.g. `2 "A"`)

**Reply by senhalil** [reply]
Yeah, it would require plurals for the keywords -- and let me tell you, I had the "chance" to look at that problem recently and it is not really evident to do it programmatically for any word  (fortunately there were libraries); however, I was thinking for the reserved keywords, since there are not many of them, it might even be okay to "hard-code" pluralize them.

For literals, I was leaning on `2 "A"s`, `some "A"s` kind of logic -- i.e., if the multiplier/coefficient is greater than 1, then `s` itself is treated part of the syntax `" "s`.
